point to full list of compilation options (#773)
root-project/web: manual/root_macros_and_shared_libraries/index.md

The `+` option compiles the code and generates a shared library `MyScript_C.so` (`MyScript_C.dll` on Windows).
The `+` command rebuilds the library only if the ROOT macro or any of the files it includes

> are newer than the library. To force recompiling the library in any case, use `++`.

Reviewer: ```suggestion
are newer than the library. To force a recompilation of the library, use `++`.
```
